scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -201,108 +247,221 @@ def _highly_variable_genes_single_batch(
     A DataFrame that contains the columns
     `highly_variable`, `means`, `dispersions`, and `dispersions_norm`.
     """
-    X = adata.layers[layer] if layer is not None else adata.X
+    data = _get_obs_rep(adata, layer=layer)
```

Comment:
It doesn’t, the changes are entirely localized to the next block. The reason is that it’s not X, it’s `X` or a layer.

Reply:
I don't know that I find it confusing that `X` is a variable that could refer to `adata.X` or something in `adata.layers`. I do think that `X` is a nicer variable name than `data` since it carries more assumptions around it (like the shape) which hold for this function, and fits with the scikit-learn convention. `data` is very generic.